File: pdfhtmlrss_platform_web/src/dto/SignatureVerificationReport.ts
export default interface SignatureVerificationReport {
    isSigned: Boolean,
    signatureDate : Date | undefined,
    issuedBy : String | undefined,

    hasExternalSignatures: Boolean
    hasRSSPAdESSignature: Boolean,
    hasRSSXMLSignature : Boolean,

    externalSignaturesViolated: Boolean,
    rssPAdESViolated : Boolean,
    rssXMLViolated : Boolean,

    rssPAdESAlgorithm : String | undefined,
    rssXMLAlgorithm : String | undefined,
}

export function isSignatureValid(report : SignatureVerificationReport) {
    return (!report.hasExternalSignatures || !report.externalSignaturesViolated)
        && (!report.hasRSSPAdESSignature || !report.rssPAdESViolated)
        && (!report.hasRSSXMLSignature || !report.rssXMLViolated)
}

File: pdfhtmlrss_platform_core/src/main/kotlin/pt/unl/fct/di/pdf_html_rss_core/data/SignatureVerificationReport.kt
package pt.unl.fct.di.pdf_html_rss_core.data

import java.util.Date

data class SignatureVerificationReport(
    val isSigned: Boolean,
    val signatureDate : Date? = null,
    val issuedBy : String? = null,

    val hasExternalSignatures: Boolean = false,
    val hasRSSPAdESSignature: Boolean = false,
    val hasRSSXMLSignature : Boolean = false,

    val externalSignaturesViolated: Boolean = false,
    val rssPAdESViolated : Boolean = false,
    val rssXMLViolated : Boolean = false,

    val rssPAdESAlgorithm : String? = null,
    val rssXMLAlgorithm : String? = null,
) {
    fun isViolated() : Boolean = externalSignaturesViolated()
            || rssPAdESSignatureViolated()
            || rssXMLSignatureViolated()

    fun externalSignaturesViolated() : Boolean  = isSigned
            && hasExternalSignatures
            && externalSignaturesViolated;

    fun rssPAdESSignatureViolated() : Boolean = isSigned
            && hasRSSPAdESSignature
            && rssPAdESViolated;

    fun rssXMLSignatureViolated() : Boolean = isSigned
            && hasRSSXMLSignature
            && rssXMLViolated

    fun hasValidRSSSignature() : Boolean = isSigned
            && hasRSSPAdESSignature
            && hasRSSXMLSignature;
}